toolchains/xslt-M4/schema-gen/metaschema-v090-datatypes.json

```
@@ -0,0 +1,104 @@
+    },
+    "DayTimeDurationDatatype": {
+      "type": "string",
+      "format": "duration",
+      "pattern": "^[-+]?P([-+]?[0-9]+D)?(T([-+]?[0-9]+H)?([-+]?[0-9]+M)?([-+]?[0-9]+([.,][0-9]{0,9})?S)?)?$"
+    },
+    "DecimalDatatype": {
+      "type": "number"
+    },
+    "EmailAddressDatatype": {
+      "type": "string",
+      "format": "email",
+      "pattern": "^.+@.+$"
+    },
+    "HostnameDatatype": {
+      "$ref": "#/definitions/StringDatatype",
+      "format": "idn-hostname"
+    },
+    "IntegerDatatype": {
+      "type": "integer"
+    },
+    "IPV4AddressDatatype": {
+      "type": "string",
+      "format": "ipv4",
+      "pattern": "^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9]).){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])$"
+    },
+    "IPV6AddressDatatype": {
+      "type": "string",
+      "format": "ipv6",
+      "pattern": "^(([0-9a-fA-F]{1,4}:){7,7}[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,7}:|([0-9a-fA-F]{1,4}:){1,6}:[0-9a-fA-F]{1,4}|([0-9a-fA-F]{1,4}:){1,5}(:[0-9a-fA-F]{1,4}){1,2}|([0-9a-fA-F]{1,4}:){1,4}(:[0-9a-fA-F]{1,4}){1,3}|([0-9a-fA-F]{1,4}:){1,3}(:[0-9a-fA-F]{1,4}){1,4}|([0-9a-fA-F]{1,4}:){1,2}(:[0-9a-fA-F]{1,4}){1,5}|[0-9a-fA-F]{1,4}:((:[0-9a-fA-F]{1,4}){1,6})|:((:[0-9a-fA-F]{1,4}){1,7}|:)|[fF][eE]80:(:[0-9a-fA-F]{0,4}){0,4}%[0-9a-zA-Z]{1,}|::([fF]{4}(:0{1,4}){0,1}:){0,1}((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9]).){3,3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9])|([0-9a-fA-F]{1,4}:){1,4}:((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9]).){3,3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9][0-9]|[0-9]))$"
+    },
+    "MarkupLineDatatype": {
+      "type": "string",
+      "pattern": "^[^\n]+$"
+    },
+    "MarkupMultilineDatatype": {
+      "type": "string"
+    },
+    "NonNegativeIntegerDatatype": {
+      "$ref": "#/definitions/IntegerDatatype",
+      "minimum": 0
+    },
+    "PositiveIntegerDatatype": {
+      "$ref": "#/definitions/IntegerDatatype",
+      "minimum": 1
+    },
+    "StringDatatype": {
+      "type": "string",
+      "pattern": "^\\S(.*\\S)?$"
+    },
+    "TokenDatatype": {
+      "type": "string",
+      "pattern": "^(\\p{L}|_)(\\p{L}|\\p{N}|[.\\-_])*$"
+    },
+    "URIDatatype": {
+      "type": "string",
+      "format": "uri",
+      "pattern": "^[a-zA-Z][a-zA-Z0-9+\\-.]+:.+$"
+    },
+    "URIReferenceDatatype": {
+      "type": "string",
+      "format": "uri-reference"
+    },
+    "UUIDDatatype": {
+      "type": "string",
+      "description": "A type 4 ('random' or 'pseudorandom') or type 5 UUID per RFC 4122.",
+      "pattern": "^[0-9A-Fa-f]{8}-[0-9A-Fa-f]{4}-[45][0-9A-Fa-f]{3}-[89ABab][0-9A-Fa-f]{3}-[0-9A-Fa-f]{12}$"
+    },
+    "YearMonthDurationDatatype": {
+      "type": "string",
+      "format": "duration",
+      "pattern": "^[-+]?P([-+]?[0-9]+Y)?([-+]?[0-9]+M)?([-+]?[0-9]+W)?([-+]?[0-9]+D)?$"
+    }
+  }
+}
```

Need to have a blank line at the end of the file.

```suggestion
}

```